src/Uno.UI/UI/Xaml/Controls/Image/Image.cs

```diff
@@ -56,14 +56,35 @@ public static new class TraceProvider
 			public const int Image_SetImageStop = 6;
 		}
 
-		protected virtual void OnImageFailed(ImageSource imageSource)
+		private protected void OnImageFailed(ImageSource imageSource, Exception exception)
```

Comment:
we could just avoid the null check... there is little point to do a branching here

Reply:
@Xiaoy312 I'm quite afraid `exception` could be null in this code path

Reply:
$"It is perfectly fine to print a null exception: {(Exception)null}" // for the logging parts